spec/models/appointment_spec.rb

```diff
@@ -0,0 +1,87 @@
+require 'rails_helper'
+
```

Comment:
The tests are passing, Great effort to implement all these tests. I would suggest in the future using fewer it blocks as it slows down the test run time.